apps/openssl.c

```diff
@@ -183,6 +185,33 @@ static void cleanup_trace(void)
     sk_tracedata_pop_free(trace_data_stack, tracedata_free);
 }
 
+static void setup_trace_category(int category)
+{
+    BIO *channel;
+    tracedata *trace_data;
+
+    if (OSSL_trace_enabled(category))
+        return;
+
+    channel = BIO_push(BIO_new(apps_bf_prefix()),
+                            dup_bio_err(FORMAT_TEXT));
+    trace_data = OPENSSL_zalloc(sizeof(*trace_data));
+
+    if (trace_data == NULL
+        || (trace_data->bio = channel) == NULL
+        || OSSL_trace_set_callback(category, internal_trace_cb,
+                                   trace_data) == 0
+        || sk_tracedata_push(trace_data_stack, trace_data) == 0) {
+
+        fprintf(stderr,
```

Comment:
Is it too early in the setup to use the error BIO here?

Reply:
I'm not sure. Since I only moved the code from here

https://github.com/openssl/openssl/blob/abf76808ee9c0a6b5861ef7da1c577a85e8b8172/apps/openssl.c#L206-L208

I'll pass the question on to @levitte.